tsl/src/chunk_api.c

```diff
@@ -1627,3 +1627,63 @@ chunk_api_get_chunk_relstats(PG_FUNCTION_ARGS)
 {
 	return chunk_api_get_chunk_stats(fcinfo, false);
 }
+
+TS_FUNCTION_INFO_V1(ts_chunk_get_dimension_slices);
+Datum
+ts_chunk_get_dimension_slices(PG_FUNCTION_ARGS)
+{
+	Oid relid = PG_ARGISNULL(0) ? InvalidOid : PG_GETARG_OID(0);
+	const Chunk *chunk = ts_chunk_get_by_relid(relid, false);
+	Hypertable *ht;
+
+	if (NULL == chunk)
+		ereport(ERROR,
+				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+				 errmsg("must be a hypertable or chunk")));
+
+	ht = ts_hypertable_get_by_id(chunk->fd.hypertable_id);
+	PG_RETURN_TEXT_P(cstring_to_text(chunk_api_dimension_slices_json(chunk, ht)));
+}
+
+Datum
+chunk_create_empty_table(PG_FUNCTION_ARGS)
+{
+	const Oid hypertable_relid = PG_GETARG_OID(0);
+	Jsonb *const slices = PG_GETARG_JSONB_P(1);
+	const char *const schema_name = PG_GETARG_CSTRING(2);
+	const char *const table_name = PG_GETARG_CSTRING(3);
+	Cache *const hcache = ts_hypertable_cache_pin();
+	Hypertable *const ht = ts_hypertable_cache_get_entry(hcache, hypertable_relid, CACHE_FLAG_NONE);
+	Hypercube *hc;
+	const char *parse_err;
+	Chunk *chunk;
+	AclResult acl_result;
+
+	Assert(!PG_ARGISNULL(0));
```

Comment:
I suggest adding not that we can assert non-nullness because the function is `STRICT`. Although I really wonder if we want `STRICT` in this case as opposed to generating a useful error? With `STRICT` this function will succeed on NULL input and return a NULL result, which then might need handling by the caller as opposed to failing with a descriptive error.

Reply:
I don't really follow why it is important to invest in descriptive error messages for NULL values in this internal function. Can you elaborate why it is important in the context of implementing copy/move functionality and the function being internal?

Reply:
See separate PR https://github.com/timescale/timescaledb/pull/3149